<?xml version="1.0" encoding="utf-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <link rel="stylesheet" href="style/base.css" type="text/css" />
</head>
<body>
    <div class="document">
        <!--div class="donemessagelarge"-->
		<strong>RetroTMiner:
		A Galaxy based system for identifying retrotransposons in genome</strong>
            <hr>
	    <code>RetroTMiner is a Galaxy workflow system for MGEScan which is identifying long terminal repeats (LTR) and non-LTR retroelements in eukaryotic genomic sequences. With a Galaxy workflow toolkit, input genome sequences are easily provided by UCSC Table Browser, ENA Browser or local storage.  HMMER 3.1b1 and gff3 are applied to enhance data processing and visualization of genome sequence data from certain data formats. </code>
	    
        </div>
	<br/>
	<b> NEW IMAGE & DESCRIPTION GO HERE (task 1) </b>
        <img src="images/noodles.png" alt="WWFSMD?" style="display: block; margin-left: auto; margin-right: auto;" />
	<hr/>
	<h1> QUICK START (task 5)</h1>
	<b> tutorial goes here </b>
	<br/>
	<h1>Citation</h1>
	<b>citation goes here</b>
	<br/>
	<h1>References (task 7)</h1>
	<b> 1. MGEScan-non-LTR: computational identification and classification of autonomous non-LTR retrotransposons in eukaryotic genomes. Rho M, Tang H. </b>
	<b> 2. LTR retroelements in the genome of Daphnia pulex. Mina Rho, Sarah Schaack, Xiang Gao, Sun Kim, Michael Lynch and Haixu Tang</b>
	<br/>
	<code>

		    Transposable elements (TEs), also called mobile genetic elements (MGE), have been found in most eukaryotic genomes. TEs often constitute a significant portion of the eukaryotic genome (e.g., 80% of the maize, 45% of the human, and 5.3% of the fruit fly genome) and play important roles in shaping its structure. Because they can transpose from one location to another within the genome or across genomes, the identification of TEs and the analysis of their dynamics are important for a better understanding of the structure and evolution of both genomes and TEs themselves. Thus, we developed a computational method MGEScan to identify long terminal repeats (LTR) and non-LTR retroelements in eukaryotic genomic sequences. MGEScan-LTR is a de novo method to identify LTR retroelements, an important class of TEs that transpose through reverse transcription of RNA intermediates. Intact LTR retroelements were identified using multiple empirical rules: similarity of a pair of LTRs at the both ends, the structure of internal regions (IRs), di(tri)-nucleotides at flanking ends, and target site duplications (TSDs). Â The intact elements identified were clustered into families based on the sequence similarity of LTRs between elements (>80%). These frameworks were applied to indentify a large number of novel elements, which were subsequently analyzed to estimate the evolutionary history and relationships of TEs. MGEScan-nonLTR is a computational approach inspired by a generalized hidden Markov model (GHMM) to identify non-LTR retroelements in genomic sequences. To model common features of non-LTR retroelements in a large variety of genomes, we built a model consisting of twelve super-states, each corresponding to a different clade (for example, I, Jockey, and R1). Each super-state consists of one to three states, corresponding to protein domains and linker regions encoded by the non-LTR retroelements. To evaluate the scores for the state of protein domains and inter-domain region, we adopted two probabilistic models, a profile HMM (for the protein domains) and a Gaussian Bayes classifier (for the linker regions). MGEScan-nonLTR was tested on the genome sequences of four eukaryotic organisms, Drosophila melanogaster, Daphnia pulex, Ciona intestinalis, and Strongylocentrotus purpuratus. Notably, for the D. pulex genome, MGEScan-nonLTR found a significantly larger number of elements than did RepeatMasker, using the current version of the RepBase Update library.</code>
        <a target="_blank" class="reference" href="http://galaxyproject.org/">Galaxy</a> is an open, web-based platform for data intensive biomedical research.  The <a target="_blank" class="reference" href="http://wiki.galaxyproject.org/GalaxyTeam">Galaxy team</a> is a part of <a target="_blank" class="reference" href="http://www.bx.psu.edu">BX</a> at <a target="_blank" class="reference" href="http://www.psu.edu">Penn State</a>, and the <a target="_blank" class="reference" href="http://www.biology.emory.edu/">Biology</a> and <a target="_blank" class="reference" href="http://www.mathcs.emory.edu/">Mathematics and Computer Science</a> departments at <a target="_blank" class="reference" href="http://www.emory.edu/">Emory University</a>.  The <a target="_blank" class="reference" href="http://galaxyproject.org/">Galaxy Project</a> is supported in part by <a target="_blank" class="reference" href="http://www.genome.gov">NHGRI</a>, <a target="_blank" class="reference" href="http://www.nsf.gov">NSF</a>, <a target="_blank" class="reference" href="http://www.huck.psu.edu">The Huck Institutes of the Life Sciences</a>, <a target="_blank" class="reference" href="http://www.ics.psu.edu">The Institute for CyberScience at Penn State</a>, and <a target="_blank" class="reference" href="http://www.emory.edu/">Emory University</a>.</p>

    </div>
</body>
</html>
